### PR TITLE
Update Terraform providers

### PR DESCRIPTION
**Pull request by meyskens**
# Update Terraform providers

This updates all Terraform providers which we use to the latest versions.
It also adds 2 bootkube dependancies to the find updates script.

# How to use

Successfully apply a cluster

# Testing done

Working on testing, first waiting on the e2e to do some obvious catches

### CODE REVIEW

File: assets/terraform-modules/bootkube/versions.tf [hunk 1]
```diff
@@ -5,15 +5,15 @@ terraform {
...
+      version = "2.1.0"
     }
     template = {
       source  = "hashicorp/template"
       version = "2.2.0"
     }
     tls = {
       source  = "hashicorp/tls"
-      version = "2.2.0"
+      version = "3.1.0"
```
Comment (surajssd) [inline assets/terraform-modules/bootkube/versions.tf:16]: This commit mentions only about Local but also updates TLS.

**Review by surajssd — COMMENTED** [review]
Just one nit, rest LGTM

File: assets/terraform-modules/bootkube/versions.tf (hunk 1 above)
Comment (meyskens) [reply assets/terraform-modules/bootkube/versions.tf:16]: Aha forgot that, fixed it!

**Comment by meyskens** [comment]
Tinkerbell is red but that is fixed in #1243

**Review by surajssd — APPROVED** [review]
LGTM

**Pull request merged**